docs/contracts/v4/concepts/02-pool-manager-and-pool.mdx

```diff
@@ -0,0 +1,175 @@
+---
+id: pool-manager-and-pool-initialization
+title: Pool Manager and Pool Initialization
+sidebar_position: 1
+---
+
+# Architecture
+In Uniswap v3, each pool has its own contract instance, which makes making pools and doing swaps in multiple pools costly
+
+![High Level Architecture](./images/01_Pool_Initialization/HighLevelArchitecture.png)
+
+In v4, all pools are kept in a single contract instance, saving a lot of important gas because tokens don’t have to move between different pool contracts during swaps. Early calculations say that v4 will make the gas cost of creating pools 99% less. Hooks offer unlimited choices and the single contract lets you easily move through all these choices.
```

Comment:
```suggestion
In v4, all pools are kept in a single contract to provide substantial gas savings. Early calculations say that v4 will make the gas cost of creating pools 99% less.
```

Don't think the singleton design helps navigate the infinite world of hooks, so dont want to mislead here